maestrowf/datastructures/yamlspecification.py

```diff
@@ -95,22 +95,34 @@ def load_specification(cls, path):
         try:
             # Load the YAML spec from the file.
             with open(path, 'r') as data:
-                try:
-                    spec = yaml.load(data, yaml.FullLoader)
-                except AttributeError:
-                    logger.warning(
-                        "*** PyYAML is using an unsafe version with a known "
-                        "load vulnerability. Please upgrade your installation "
-                        "to a more recent version! ***")
-                    spec = yaml.load(data)
-
+                specification = cls.load_specification_from_stream(data)
         except Exception as e:
             logger.exception(e.args)
             raise
 
+        return specification
+
+    @classmethod
+    def load_specification_from_stream(cls, stream):
+        """
+        Load a study specification.
+
+        :param stream: Raw text stream to study YAML specification data.
+        :returns: A specification object containing the information from the
+        passed stream.
+        """
+
+        try:
+            spec = yaml.load(stream, yaml.FullLoader)
+        except AttributeError:
+            logger.warning(
+                "*** PyYAML is using an unsafe version with a known "
+                "load vulnerability. Please upgrade your installation "
+                "to a more recent version! ***")
+            spec = yaml.load(stream)
+
         logger.debug("Loaded specification -- \n%s", spec["description"])
         specification = cls()
-        specification.path = path
```

Comment:
Sorry, one thing I neglected to comment on yesterday-- do we want to move this line up to line 98? So that the `path` attribute still exists? Right now, it looks as if the `path` attribute is gone, which is breaking some of my code.

Reply:
Good catch -- I thought I had moved it, but it appears I didn't. Keep an eye out for a quick fix soon.

Reply:
Another thought -- if you're using the stream form of the API, that variable can't be filled by the method. That opens you up to `YAMLSpecification` instances without a populated path member variable.

Reply:
Right, in that case it would be `None`, or the responsibility of the person that called `load_specification_from_stream` to populate it?

Reply:
I'll have it populated to `None` so it's explicit that it doesn't have one. That seems to be the best answer here to me.

Reply:
There's now a quick fix in `develop`.

Reply:
Thanks! This appears to fix everything.